.github/workflows/tests.yml

```diff
@@ -52,3 +52,30 @@ jobs:
         run: docker load < docker-image.tar
       - name: Lint and Type Checking
         run: docker run --rm amp lint
+
+  push:
+    name: Upload the Docker image to Docker Hub
+    runs-on: ubuntu-latest
+    # We only want to publish a new image when the master branch changes, not
+    # for PRs and only if tests finish successfully.
+    needs: tests
+    if: github.event_name == 'push'
```

Comment:
I guess this is copied from Azafea, so preëxisting and not a blocker, but I have in the past found it useful to be able to manually trigger a GitHub Actions run without pushing a commit. e.g. if I had absent-mindedly merged this while the secrets are (presumably) not present in the repo, this step would have failed, and the only way to retrigger it would be (I think?) to push a new commit.

So you could imagine adding a `workflow_dispatch:` block to the `on:` at the top and adjusting this condition accordingly.

Reply:
Good call. I had that in mind but forgot about it by the time I actually got here.